README.md

```diff
@@ -1,2 +1,151 @@
+sufficient.
+
+    $ mkdir build
+    $ cd build
+
+The next step is to run CMake to configure the project.  Running CMake is the
+equivalent to running `./configure` with autotools. 
+
+    $ cmake ../
+
+Once CMake is configured, building the library is as simple as typing 'make'.
+
+    $ make
+
+If you wish to install mlboard to `/usr/local/include/mlboard/`, `/usr/local/lib/`,
+and `/usr/local/bin/`, make sure you have root privileges (or write permissions 
+to those three directories), and simply type
+
+    $ make install
+
+and the mlboard headers are found in `/usr/local/include/mlpack/`.
+
+### 2. Usage
+
+To generate summary data you can make use of `fileWriter` class and `SummaryWriter` class.
+
+For creating a summary, you have to call the specific summary type function from `SummaryWriter` class, for example to log a scaler summary you can call the scaler function as `mlboard::SummaryWriter<mlboard::fileWriter>::scalar(tag,step,value,filewriterobject);`. 
+
+Irrespective of the summary type, you always have to pass the filewriter object that is responsible for first creating events using this summary and then putting that into a queue and then finally writing those events into the file through the logger, which is running asynchronously.
+
+Following is a snippet that would log scaler for 4 steps in temp directory.
+
+You can compile the following snippet of code using : `g++ main.cpp -lproto -lprotobuf -lpthread` 
+
+```
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+using namespace std;
+using namespace mlboard;
+int main()
+{
+    GOOGLE_PROTOBUF_VERIFY_VERSION;
+    // creating a filewriter object that is repnsible for logging the summary.
+    fileWriter f1("temp");
+    std::chrono::time_point<std::chrono::system_clock> start, end; 
+    start = std::chrono::system_clock::now(); 
+    
+    // Creating a scaler summary
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",1,1.1,f1);
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",2,1.2,f1);
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",3,1.4,f1);
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",5,1.5,f1);
+    end = std::chrono::system_clock::now(); 
+  
+    std::chrono::duration<double> elapsed_seconds = end - start; 
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+  
+    std::cout << "finished computation at " << std::ctime(&end_time) 
+              << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
+
+    // This will allow you to indicate that you have logged all your data.
+    f1.close();
+    google::protobuf::ShutdownProtobufLibrary();
+}
+```
+
+You can then use tensorboard to visualize the scaler by using a simple command: `tensorboard --logdir .`
+
+The above snippet of code would print the time taken for the scaler to log
+
+```
+elapsed time: 40.0163s
```

Comment:
2 sec is too much time in my opinion and hence I choose 2 millisec, I you still want sec it as simple as using boost::second instead of std::chrono::seconds instead of std::chrono::milliseconds 

Let me know what you want

I agree that tensorflow has 2 sec , but I think cpp is faster than python and hence that time should also change and not follow same time

So let me know what you want

Reply:
Yes, you're correct according to language the time should go down. We need to ensure that it is not very much small. I think 2 milisec is very small. Basically we would be logging Metrics of ML models I think hardly any ML model would complete it's one step in 2 milisec.

Also, in tensorflow the default is 120000 milisec(2 mins). So I think we should reduce to 120000 / 10 = 120000 assuming c++ is 10x faster. I am fine with 2 secs all up to you.

https://www.tensorflow.org/api_docs/python/tf/summary/create_file_writer

Reply:
Let's be in middle 5 sec :)